File: qn1.py
'''
Find a pair in an array whose sum is equal to given number
'''
def find_pair(array,given_sum):
    d = {}
    for x in array:
        other = given_sum - x
        if other in d:
            print("the no. are : " ,x, other)
        else:
            d[x] = True
            
IF __NAME__ == '__MAIN__':
    
            
#array = [0,1,2,3,4,5]
#find_pair(array,5)

File: arrays/qn1.c
/*
Find a pair in an array whose sum is equals to given number
*/
#include<stdio.h>

void findPair(int *a,int size,int k){
    int hash[100] = {0};
    int curr = 0,other;
    char i,j;
    
    for(i=0;i<size;i++){
        if(hash[curr]==1){
            printf("\nThe pair is:%d and %d",curr,other);
        }
        curr = a[i];
        other = k - a[i];
        hash[other]=1;
    }
}

int main(){
    int size;
    int a[10],i,k;
    printf("\nEnter size of array: ");
    scanf("%d",&size);
    printf("\nEnter Elements of array:");
    for(i=0;i<size;i++){
        scanf("%d",&a[i]);
    }
    printf("Enter sum: ");
    scanf("%d",&k);
    findPair(a,size,k);
    return;
}